### ISSUE TITLE
Replace GET parameter request authentication in GitHub site handler

### ISSUE THREAD

**Issue by koobs** [freebsd/portscout#8]
GitHub is deprecating [1] authentication via URL query parameters:

```
Please use Basic Authentication instead as using OAuth credentials in query parameters
has been deprecated and will be removed July 1st, 2020.
```

We need to switch our GitHub Site Handler to use HTTP Basic Auth, or another supported authentication method, whichever requires the least code changes

[1] https://developer.github.com/changes/2019-11-05-deprecated-passwords-and-authorizations-api/#authenticating-using-query-parameters

**Comment by grembo** [comment]
Github recommends using the oauth-token in an Authorization header instead of using Basic Auth with username+token (using the account password will be deprecated in api calls). Testing also showed that the username is ignored when using basic auth, as long as one is provided. Only the token is used for authentication/authorization.

See also: https://developer.github.com/v3/#oauth2-token-sent-in-a-header

**Comment by grembo** [comment]
This was addressed in #9 and #13
